cx/slack/server.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * TODO(developer): 
+ * Add your service key to the current folder.
+ * Uncomment and fill in these variables.
+ */
+// const slackSigningSecret = '...';
+// const slackToken = 'xoxb-...';
+// const projectId = 'my-project';
+// const locationId = 'global';
+// const agentId = 'my-agent';
+// const languageCode = 'en';
+
+const { WebClient } = require('@slack/web-api');
+const { createEventAdapter } = require('@slack/events-api');
+
+const slackEvents = createEventAdapter(slackSigningSecret);
+const slackClient = new WebClient(slackToken);
+
+// Imports the Google Cloud Some API library
+const {SessionsClient} = require('@google-cloud/dialogflow-cx');
+/**
+ * Example for regional endpoint:
+ *   const locationId = 'us-central1'
+ *   const client = new SessionsClient({apiEndpoint: 'us-central1-dialogflow.googleapis.com'})
+ */
+const client = new SessionsClient({apiEndpoint: locationId + '-dialogflow.googleapis.com'});
+
+/**
+ * Converts Slack request to a detectIntent request.
+ */
+function slackToDetectIntent(slackRequest,sessionPath){
+    const request = {
+        session: sessionPath,
+        queryInput: {
+            text: {
+                text: slackRequest.text,
+            },
+            languageCode,
+        },
+    };
+
+    return request;
+}
+
+/**
+ * Converts detectIntent response to a Slack message request. 
+ */
+function detectIntentToSlackMessage(response,id){
```

Comment:
The id can also be user id depending on where the message is coming from. If the message is from an im it is a user id but if  not it will be a channal id.

Reply:
But it seems to always be `channel_id` from the code?
```
const request = {
    channel: id,
    text: agentResponse
};
```

Reply:
Yes but since the integration needs to also support ims the so can also be set to user.id, but the slack request structure labels both as channals so I can change it.

Reply:
Changed the variable name.

Reply:
Changed the veriable name.